consensus/checks.go

```diff
@@ -166,3 +184,70 @@ func (consensus *Consensus) onPreparedSanityChecks(
 
 	return true
 }
+
+func (consensus *Consensus) viewChangeSanityCheck(msg *msg_pb.Message) bool {
+	senderKey, err := consensus.verifyViewChangeSenderKey(msg)
+	if err != nil {
+		consensus.getLogger().Error().Err(err).Msgf(
+			"[%s] VerifySenderKey Failed",
+			msg.GetType().String(),
+		)
+		return false
+	}
+	if err := verifyMessageSig(senderKey, msg); err != nil {
+		consensus.getLogger().Error().Err(err).Msgf(
+			"[%s] Failed To Verify Sender's Signature",
+			msg.GetType().String(),
+		)
+		return false
+	}
+	return true
+}
+
+func (consensus *Consensus) onViewChangeSanityCheck(recvMsg *FBFTMessage) bool {
+	// TODO: if difference is only one, new leader can still propose the same committed block to avoid another view change
```

Comment:
how relevant is this todo , what about associated formal github issue

Reply:
this was old TODO, I didn't add. It is still relevant though.

Reply:
If still relevant then let’s make explicit issue for it